src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                    { SyntaxKind.NotEqualsExpression, (SyntaxKind.EqualsExpression, SyntaxKind.EqualsEqualsToken) },
+                    { SyntaxKind.LessThanExpression, (SyntaxKind.GreaterThanOrEqualExpression, SyntaxKind.GreaterThanEqualsToken) },
+                    { SyntaxKind.LessThanOrEqualExpression, (SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanToken) },
+                    { SyntaxKind.GreaterThanExpression, (SyntaxKind.LessThanOrEqualExpression, SyntaxKind.LessThanEqualsToken) },
+                    { SyntaxKind.GreaterThanOrEqualExpression, (SyntaxKind.LessThanExpression, SyntaxKind.LessThanToken) },
+                };
+
+        protected override SyntaxNode GetIfStatement(TextSpan textSpan, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            var ifStatement = token.GetAncestor<IfStatementSyntax>();
+            if (ifStatement == null || ifStatement.Else == null)
+            {
+                return null;
+            }
+
+            var span = TextSpan.FromBounds(ifStatement.GetFirstToken().Span.Start, ifStatement.CloseParenToken.Span.End);
+            if (!span.IntersectsWith(textSpan))
+            {
+                return null;
+            }
+
+            return ifStatement;
+        }
+
+        protected override SyntaxNode GetRootWithInvertIfStatement(Workspace workspace, SemanticModel model, SyntaxNode ifStatement, CancellationToken cancellationToken)
+        {
+            var ifNode = (IfStatementSyntax)ifStatement;
+
+            // For single line statment, we swap the TrailingTrivia to preserve the single line
+            StatementSyntax newIfNodeStatement = null;
+            ElseClauseSyntax newElseStatement = null;
+
+            var isMultiLine = ifNode.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia);
+            if (isMultiLine)
+            {
+                // multiline if
+                newIfNodeStatement = ifNode.Else.Statement.Kind() != SyntaxKind.Block
+                ? SyntaxFactory.Block(ifNode.Else.Statement)
+                : ifNode.Else.Statement;
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement);
+            }
+            else
+            {
+                // singleline if
+                var elseTrailingTrivia = ifNode.Else.GetTrailingTrivia();
+                var ifTrailingTrivia = ifNode.Statement.GetTrailingTrivia();
+                newIfNodeStatement = ifNode.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement).WithTrailingTrivia(elseTrailingTrivia);
+            }
+
+            ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken))
+                .WithStatement(ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block ? SyntaxFactory.Block(newIfNodeStatement) : newIfNodeStatement)
```

Comment:
please pull out hte argument to WithStatement into its own variable.  it's way too complex and confusing to read in this form.